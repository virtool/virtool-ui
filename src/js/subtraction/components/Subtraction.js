import React from "react";
import { Route, Switch } from "react-router-dom";
import { Container, NarrowContainer } from "../../base";
import { SubtractionFileManager } from "./FileManager";
import SubtractionCreate from "./Create";
import SubtractionDetail from "./Detail/Detail";
import SubtractionList from "./List";

const Subtraction = () => (
    <Container>
        <NarrowContainer>
            <Switch>
                <Route path="/subtractions" component={SubtractionList} exact />
                <Route path="/subtractions/files" component={SubtractionFileManager} />
                <Route path="/subtractions/create" component={SubtractionCreate} />
                <Route path="/subtractions/:subtractionId" component={SubtractionDetail} />
            </Switch>
        </NarrowContainer>
    </Container>
);

export default Subtraction;
