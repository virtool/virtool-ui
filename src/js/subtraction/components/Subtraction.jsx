import React from "react";
import { Route, Switch } from "react-router-dom";
import { Container, ContainerNarrow } from "../../base";
import SubtractionCreate from "./Create";
import SubtractionDetail from "./Detail/Detail";
import { SubtractionFileManager } from "./FileManager";
import SubtractionList from "./List";

const Subtraction = () => (
    <Container>
        <ContainerNarrow>
            <Switch>
                <Route path="/subtractions" component={SubtractionList} exact />
                <Route path="/subtractions/files" component={SubtractionFileManager} />
                <Route path="/subtractions/create" component={SubtractionCreate} />
                <Route path="/subtractions/:subtractionId" component={SubtractionDetail} />
            </Switch>
        </ContainerNarrow>
    </Container>
);

export default Subtraction;
