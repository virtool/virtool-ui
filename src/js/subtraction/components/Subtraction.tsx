import { Container, ContainerNarrow } from "@base";
import React from "react";
import { Route, Switch } from "react-router-dom";
import SubtractionDetail from "./Detail/SubtractionDetail";
import { SubtractionFileManager } from "./SubtractionFileManager";
import SubtractionList from "./SubtractionList";

/**
 * The subtraction view
 */
export default function Subtraction() {
    return (
        <Container>
            <ContainerNarrow>
                <Switch>
                    <Route path="/subtractions" component={SubtractionList} exact />
                    <Route path="/subtractions/files" component={SubtractionFileManager} />
                    <Route path="/subtractions/:subtractionId" component={SubtractionDetail} />
                </Switch>
            </ContainerNarrow>
        </Container>
    );
}
