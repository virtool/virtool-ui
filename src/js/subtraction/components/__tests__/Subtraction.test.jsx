import { shallow } from "enzyme";
import React from "react";
import { describe, expect, it } from "vitest";
import Subtraction from "../Subtraction";

describe("<Subtraction />", () => {
    it("should render", () => {
        const wrapper = shallow(<Subtraction />);
        expect(wrapper).toMatchSnapshot();
    });
});
