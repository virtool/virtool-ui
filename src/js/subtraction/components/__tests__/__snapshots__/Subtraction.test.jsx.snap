// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Subtraction /> > should render 1`] = `
<Container>
  <ContainerNarrow>
    <Switch>
      <Route
        component={
          {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        exact={true}
        path="/subtractions"
      />
      <Route
        component={[Function]}
        path="/subtractions/files"
      />
      <Route
        component={
          {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        path="/subtractions/create"
      />
      <Route
        component={
          {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
          }
        }
        path="/subtractions/:subtractionId"
      />
    </Switch>
  </ContainerNarrow>
</Container>
`;
